src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMKeyValue.java

```diff
@@ -261,9 +261,26 @@ public boolean equals(Object obj) {
                         && rsaKey.getModulus().equals(otherRSAKey.getModulus());
             } catch (KeyException ke) {
                 // no practical way to determine if the keys are equal
-                return false;
+                return super.equals(obj);
```

Comment:
Actually, I take back my comment here - returning false as originally is ok (and better) since it already checked if the objects were the same on line 243.